modules/message/src/main/JsonView.scala

```diff
@@ -0,0 +1,44 @@
+package lila.message
+
+import play.api.libs.json._
+import play.api.mvc.Result
+import play.api.mvc.Results._
+import scala.concurrent.duration._
+
+import lila.common.PimpedJson._
+import lila.user.User
+import scala.concurrent.{ Future }
+import lila.common.paginator._
+
+final class JsonView() {
+
+  def inbox(me: User, threads: Paginator[Thread]): Result =
+    Ok(Json.obj(
+      "threads" -> threads.currentPageResults.map { thread => inboxItem(me, thread) }
```

Comment:
serialize the entire paginator instead, like this: https://github.com/ornicar/lila/blob/master/modules/api/src/main/UserGameApi.scala#L21

Here I map the paginator results to convert them to JSON: https://github.com/ornicar/lila/blob/master/modules/api/src/main/UserApi.scala#L22-L24